include/circt/Dialect/HW/HWStructure.td

```diff
@@ -112,6 +112,10 @@ def HWModuleOp : HWModuleOpBase<"module",
       return getModuleOutputPort(*this, i);
     }
 
+    /// Adds output ports to the module with the specified names and rewrites
+    /// the output op to return the associated values.
+    void addOutputs(SmallVectorImpl<std::pair<StringAttr, Value>> &outputs);
```

Comment:
How about making this take a `ArrayRef<std::pair<...>>` such that you can pass in other stuff besides just `SmallVector`?